scheduler.go

```diff
@@ -86,124 +86,148 @@ func (s *Scheduler) ChangeLocation(newLocation *time.Location) {
 }
 
 // scheduleNextRun Compute the instant when this Job should run next
-func (s *Scheduler) scheduleNextRun(j *Job) {
-	j.Lock()
-	defer j.Unlock()
+func (s *Scheduler) scheduleNextRun(job *Job) {
+	job.Lock()
+	defer job.Unlock()
 	now := s.time.Now(s.loc)
 
-	if j.startsImmediately {
-		j.nextRun = now
-		j.startsImmediately = false
+	if job.startsImmediately {
+		job.nextRun = now
+		job.startsImmediately = false
 		return
 	}
 
-	// delta represent the time slice used to calculate the next run
-	// it can be the last time ran by a job, or time.Now() if the job never ran
-	var delta time.Time
-	if j.neverRan() {
-		if !j.nextRun.IsZero() { // scheduled for future run, wait to run at least once
-			return
-		}
-		delta = now
-	} else {
-		delta = j.lastRun
+	if job.neverRan() && !job.nextRun.IsZero() {
+		return // scheduled for future run and should skip scheduling
 	}
+	job.lastRun = now
+
+	durationToNextRun := s.durationToNextRun(*job)
+	job.nextRun = job.lastRun.Add(durationToNextRun)
+}
 
-	switch j.unit {
+func (s *Scheduler) durationToNextRun(job Job) time.Duration {
+	lastRun := job.lastRun
+	var duration time.Duration
+	switch job.unit {
 	case seconds, minutes, hours:
-		j.nextRun = s.rescheduleDuration(j, delta)
+		duration = s.calculateDuration(job)
 	case days:
-		j.nextRun = s.rescheduleDay(j, delta)
+		duration = s.calculateDays(job, lastRun)
 	case weeks:
-		j.nextRun = s.rescheduleWeek(j, delta)
+		{
```

Comment:
nit: Do we need the curly brace here? I think Go allows us to have multi-line block of code within a `case` without having to enclose it in curly braces.

Reply:
nice catch, fixing!